### PR TITLE
implements stale update warning

### PR DESCRIPTION
**Pull request by renlord**
show warning notification when updater fails to reach update server for more than 1 day

### CODE REVIEW

File: res/values/strings.xml [hunk 1]
```diff
@@ -11,6 +11,8 @@
     <string name="notification_text_legacy">Please install the latest OS release.</string>
     <string name="notification_reboot_action">Reboot</string>
     <string name="notification_reboot_action_legacy">Reboot and install</string>
+    <string name="notification_updater_contact_fail_title">Update Stale Warning</string>
```
Comment (chirayudesai) [inline res/values/strings.xml]: How about "Failed to check for updates"

File: src/app/seamlessupdate/client/Service.java [hunk 2]
```diff
@@ -320,6 +340,9 @@ protected void onHandleIntent(final Intent intent) {
             mUpdating = false;
             PeriodicJob.scheduleRetry(this);
         } finally {
+            if (System.currentTimeMillis() - getLastUpdateCheckTime > ONEDAY) {
```
Comment (chirayudesai) [inline src/app/seamlessupdate/client/Service.java]: `getLastUpdateCheckTime()`

**Review by chirayudesai — CHANGES_REQUESTED** [review]
One more thing that could be done is show the last update check time below the "Check for updates" button.

Can show it at
https://github.com/GrapheneOS/platform_packages_apps_Updater/blob/10/src/app/seamlessupdate/client/Settings.java#L55
using
https://developer.android.com/reference/android/preference/Preference#setSummary(java.lang.CharSequence)

File: res/values/strings.xml (hunk 1 above)
Comment (renlord) [reply res/values/strings.xml]: applied your suggested changes.

File: src/app/seamlessupdate/client/Service.java (hunk 2 above)
Comment (renlord) [reply src/app/seamlessupdate/client/Service.java]: thanks for pointing out the typo

**Comment by renlord** [comment]
closes https://github.com/GrapheneOS/os_issue_tracker/issues/161

**Comment by thestinger** [comment]
Needs to be rebased @renlord. Still need to think about this though.

**Comment by renlord** [comment]
@thestinger no interest to accept this feature? or approach incorrect?

**Pull request closed without merging**